### PR TITLE
Issue #929 Get rid of toMaintenance

### PR DESCRIPTION
**Pull request by ljacomet**
* Add `Persistent(CacheManager|UserManagedCache).destroy()`
* Removed `Persistent(CacheManager|UserManagedCache).toMaintenance()`

### CODE REVIEW

File: core/src/main/java/org/ehcache/core/PersistentUserManagedEhcache.java [hunk 1]
```diff
@@ -81,30 +80,16 @@ public PersistentUserManagedEhcache(CacheConfiguration<K, V> configuration, Stor
...
-      };
       st.succeeded();
-      return maintainable;
-    } catch (RuntimeException e) {
-      throw st.failed(e);
+    } catch (Throwable t) {
+      throw st.failed(t);
     }
+    destroyInternal();
+    // TODO Exit maintenance mode once #934 is solved
```
Comment (lorban) [inline core/src/main/java/org/ehcache/core/PersistentUserManagedEhcache.java]: Another todo?

Comment (ljacomet) [inline core/src/main/java/org/ehcache/core/PersistentUserManagedEhcache.java]: Yeah, let me get rid of it and instead add comments into #934

**Comment by ljacomet** [comment]
Dropped TODO and replaced by comment in #934

**Comment by lorban** [comment]
LGTM

**Comment by ljacomet** [comment]
@chrisdennis @cljohnso @AbfrmBlr Anything to add?

**Comment by chrisdennis** [comment]
Shouldn't Maintainable be moved out of the API... is it being exposed anywhere now?

**Comment by ljacomet** [comment]
Not only it is no longer exposed, it is not even used at all. So I deleted it ... PR updated

**Pull request merged**